lib/process-services/src/lib/task-list/components/task-form/claim-task.directive.spec.ts

```diff
@@ -0,0 +1,116 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, Output, EventEmitter } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { setupTestBed } from '@alfresco/adf-core';
+import { of } from 'rxjs';
+import { TaskListService } from '../../services/tasklist.service';
+import { ProcessTestingModule } from '../../../testing/process.testing.module';
+
+describe('ClaimTaskDirective', () => {
+
+    @Component({
+        selector:  'adf-claim-test-component',
+        template: '<button adf-claim-task [taskId]="taskId" (success)="onClaim($event)">Claim</button>'
+    })
+    class TestComponent {
+        taskId = 'test1234';
+        @Output()
+        claim: EventEmitter<any> = new EventEmitter<any>();
+
+        onClaim(event) {
+            this.claim.emit(event);
+        }
+    }
+
+    let fixture: ComponentFixture<TestComponent>;
+    let taskListService: TaskListService;
+
+    setupTestBed({
+        imports: [
+          ProcessTestingModule
+        ],
+        declarations: [
+            TestComponent
+        ]
+    });
+
+    beforeEach(() => {
+        taskListService = TestBed.get(TaskListService);
+        fixture = TestBed.createComponent(TestComponent);
+        fixture.detectChanges();
+    });
+
+    it('Should be able to call claim task service', () => {
+        const claimTaskSpy = spyOn(taskListService, 'claimTask').and.returnValue(of({}));
+        const button = fixture.nativeElement.querySelector('button');
+        button.click();
+        expect(claimTaskSpy).toHaveBeenCalledWith(fixture.componentInstance.taskId);
+    });
+
+    it('Should be able to catch success event on click of claim button', async() => {
+        spyOn(taskListService, 'claimTask').and.returnValue(of({}));
```

Comment:
can you make the test less dense and respect the rule https://w3.alfresco.com/confluence/pages/viewpage.action?spaceKey=AP&title=Code+Style+E2E 05-14

Reply:
Updated